compiler/rustc_session/src/options.rs

```diff
@@ -1638,7 +1638,7 @@ options! {
         "extra arguments to append to the linker invocation (space separated)"),
     #[rustc_lint_opt_deny_field_access("use `Session::link_dead_code` instead of this field")]
     link_dead_code: Option<bool> = (None, parse_opt_bool, [TRACKED],
-        "keep dead code at link time (useful for code coverage) (default: no)"),
+        "try to generate and link some kinds of dead code (default: no)"),
```

Comment:
Is there any sense to which kinds? If there isn't, I'd just drop the qualifier. I don't think there is. As far as I'm aware, the kinds of dead code we don't emit and link are simply a random hodgepodge of cases nobody has patched yet.